Small UI Improvements based on scientist feedback

corviday opened the pull request:
This PR combines a few small UI improvements requested by scientists:

1. Different spatial organization of river basins (split mainland into multiple basins)  (basins.yaml - most of the work here was in the database)
2. Add a note that you can only view salmon populations if we also have indicator data for them (PopulationTable.js)
3. Allow users to choose whether to view indicator data averaged over the entire region, or only at the outlet point (AreaDisplay.js, OutletToggle.js, DailyDataDisplay.js, MonthlyDataDisplay.js, YearlyDataDisplay.js, useStore.js)
4. Enable logarithmic colour scaling for certain indicators only (MapControls.js, LogscaleCheckbox.js)
5. Disable logarithmic colour scaling entirely, because upon further examination, all the data files for those indicators contain zero values and cannot be viewed with logarithmic colour scaling after all - ncWMS throws an error if you ask it to generate a logarithmic image of a dataset with zeroes in it (MapControls.js, LogscaleCheckbox.js)

Since these are UI changes, please feel free to weigh in on where the UI is clunky or could be improved.

[Demo](https://services.pacificclimate.org/dev/scip/app/)

Resolves #60

rod-glover commented on src/components/MonthlyDataDisplay/MonthlyDataDisplay.js:
```diff
@@ -19,27 +19,30 @@ function MonthlyDataDisplay({
 
   useEffect(() => {
     if (region && variable && rasterMetadata) {
+      const area = viewOutletIndicators ? JSON.parse(region.outlet) : region.boundary;
```
The `area` computation is repeated 3 different places. Consider factoring it out to a function?

It's tempting to consider DRYing up the other redundancies between these data displays. That might more trouble than it's worth and then having to undo it when they diverge at some future point. But worth a moment's thought.

rod-glover commented on src/components/MapControls/MapControls.js:
```diff
@@ -121,11 +121,21 @@ function MapControls({onChange, mapDataset}) {
-                const logscale = (indicatorConfig && indicator in indicatorConfig) ?
-                    indicatorConfig[indicator].logscale : false;
-                
+                const config = indicatorConfig && indicator in indicatorConfig ?
```
I think this can be simplified to `indicatorConfig?.[indicator]`. See [optional chaining operator.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

rod-glover commented on src/components/MapControls/LogScaleCheckbox.js:
```diff
@@ -5,7 +5,26 @@
+    //    return indicatorConfig 
+    //        && mapDataset.variable in indicatorConfig 
+    //        && indicatorConfig[mapDataset.variable].logscale
+    //        && minmax.min >= .01; 
```
Again can be somewhat simplified by using optional chaining operator `?.`

rod-glover reviewed (approved):
LGTM. A few minor improvements to code. 

Some UI observations to come separately.

rod-glover commented:
UI feedback:

Controls immediately below the map:
- Look vertically overcrowded to me.
- In the [Climatology | Time | Label | Time | Climatology] control, the Label text should be wrapped in a suitable element to format it consistently with other elements.
- The opacity control isn't labelled. Kind of a mystery knob until you try it. And even then you have to know what the change you see actually stems from.

Behaviour issues:

Missing data?
  1. Select basin Haida Gwaii and an indicator
  2. The map shows a raster overlaying other parts of the province but not Haida Gwaii. 
  3. Q: Is that intended? To the naive eye (which may not be the target user group), it is confusing. The graphs all show zero -- so perhaps this is a result of missing data?

No indication when watershed selection is required.
  1. Select basin Fraser River
  2. Select a Yearly indicator.
  3. Indefinitely persisting message "Loading data..." in graph area. Looks as if there is something wrong.
  4. Select watershed.
  5. Yearly indicator data actually loads and shows.
  6. This is confusing: No indication that you have to select a watershed. And in some cases (e.g., Haida Gwaii) you _can't_ select a watershed. Need better feedback from the UI here ... probably a different message in the graph area. This could be a great place to provide instructions on what to do to cause a graph to appear.

corviday commented:
Haidi Gwaii was erroneously included on the whitelist of selectable river basins. Thank you for catching that.

corviday commented:
Thank you for the detailed feedback. 

Changes:

- Aside form the incorrect selectabilityh of Haida Gwaii, i have not been able to reproduce the data errors, unfortunately. The behaviour you describe corresponds to the back end returning an error instead of data, but I can't get it to do that. I changed the  the  "Select a watershed and an indicator to see this graph" messages to say "region" instead of "watershed", which I think will make things less confusing.
- After a bunch of fruitless fooling around with various bootstrap labels which didn't make anything less crowder, I finally made the "label" element below the map into a disabled  button.
- Added text labels for the opacity and pallet controls, which also spaces out the panel some, as I could not get them onto the same horizontal as the controls.
